### PR TITLE
Fix test_api_endtoend, use deepdiff for comparison

### PR DESCRIPTION
**Pull request by mshriver**
Deepdiff was already installed as a sub-dependency, make it directly required
Use deepdiff to provide rich feedback in the test failure of what the difference is between the api path data structures

There appears to be a regression of https://github.com/Katello/katello-selinux/pull/24 that is causing candlepin_events service to fail to start. This causes the `EndToEndTestCase::test_positive_ping` testcase to fail on 6.9 snap 1

```
setup@localhost:~/repos/robottelo (fix-api-endtoend-test*%=) % pytest -v -m build_sanity tests/foreman/endtoend/test_api_endtoend.py
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/setup/repos/robottelo/.robottelo/bin/python3.8
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, inifile: pytest.ini
plugins: forked-1.3.0, services-1.3.1, xdist-1.34.0, cov-2.10.1, mock-3.3.1, ibutsu-1.1.1
collecting ... 2020-11-12 18:05:42 - conftest - DEBUG - Collected 7 test cases
collected 7 items / 1 deselected / 6 selected                                                                                                                                                

tests/foreman/endtoend/test_api_endtoend.py::AvailableURLsTestCase::test_positive_get_links PASSED                                                                                     [ 16%]
tests/foreman/endtoend/test_api_endtoend.py::AvailableURLsTestCase::test_positive_get_status_code PASSED                                                                               [ 33%]
tests/foreman/endtoend/test_api_endtoend.py::EndToEndTestCase::test_positive_find_admin_user PASSED                                                                                    [ 50%]
tests/foreman/endtoend/test_api_endtoend.py::EndToEndTestCase::test_positive_find_default_loc PASSED                                                                                   [ 66%]
tests/foreman/endtoend/test_api_endtoend.py::EndToEndTestCase::test_positive_find_default_org PASSED                                                                                   [ 83%]
tests/foreman/endtoend/test_api_endtoend.py::EndToEndTestCase::test_positive_ping FAILED                                                                                               [100%]

```

- bulk system_purpose: https://bugzilla.redhat.com/show_bug.cgi?id=1894159
- simple content access: https://bugzilla.redhat.com/show_bug.cgi?id=1890658
- content_view import: https://projects.theforeman.org/issues/30004
- content_view export_histories: https://projects.theforeman.org/issues/29999
- content_view export_api_status: https://projects.theforeman.org/issues/30961
- products bulk and repositories verify_checksum: https://projects.theforeman.org/issues/30190
- registration: https://projects.theforeman.org/issues/30440
- scap_contents bulk upload: https://github.com/theforeman/foreman_openscap/pull/449  (Thanks @xprazak2)

### CODE REVIEW

File: tests/foreman/endtoend/test_api_endtoend.py [hunk 1]
```diff
@@ -951,6 +958,16 @@ def test_positive_get_links(self):
         #          '/katello/api/organizations/:organization_id/content_views',
         #          '/katello/api/organizations/:organization_id/content_views',
         #     ], …}
+        api_paths = response.json()['links']
+        transformed_paths = {ep: tuple(paths.values()) for ep, paths in api_paths.items()}
+        api_diff = DeepDiff(
```
Comment (mshriver) [inline tests/foreman/endtoend/test_api_endtoend.py:963]: Deepdiff covers both assertions that were made before with comparing the keys and values within the groups.

It provides a rich diff when the test case does fail, making it far easier to identify what the difference between expected and transformed data structures are.

File: tests/foreman/endtoend/test_api_endtoend.py [hunk 2]
```diff
@@ -896,6 +906,11 @@ class AvailableURLsTestCase(TestCase):
     def setUp(self):
         """Define commonly-used variables."""
         self.path = f'{settings.server.get_url()}/api/v2'
+        if is_open('BZ:1887932'):
+            missing = '/katello/api/activation_keys/:activation_key_id/subscriptions'
+            API_PATHS['subscriptions'] = tuple(
+                path for path in API_PATHS['subscriptions'] if path != missing
+            )
```
Comment (JacobCallahan) [inline tests/foreman/endtoend/test_api_endtoend.py:913]: could make this open for any future additions
```suggestion
        missing = []
        if is_open('BZ:1887932'):
            missing.append('/katello/api/activation_keys/:activation_key_id/subscriptions')
        API_PATHS['subscriptions'] = tuple(
            path for path in API_PATHS['subscriptions'] if path not in missing
        )
```

**Review by JacobCallahan — APPROVED** [review]
ACK. One non-blocking suggestion

File: tests/foreman/endtoend/test_api_endtoend.py (hunk 2 above)
Comment (mshriver) [reply tests/foreman/endtoend/test_api_endtoend.py:913]: Going to do it with a dict though, so its not just subscriptions that can be added to.

**Review by lhellebr — COMMENTED** [review]
This looks sensible but have you checked whether all those changes in API are intended? This test is there to catch changes so just fixing it to pass may just hide a true positive. Similar PR that I did before has every change documented: https://github.com/SatelliteQE/robottelo/pull/8058

**Comment by mshriver** [comment]
@lhellebr Added links to BZs and foreman issues where these endpoints were introduced.  Only one I couldn't find a source for was scap_content bulk_upload.

**Comment by lhellebr** [comment]
I think @xprazak2 should be able to help here

**Comment by xprazak2** [comment]
bulk_upload endpoint for scap content was added in https://projects.theforeman.org/issues/25387

**Review by lhellebr — APPROVED** [review]

**Pull request merged**